src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -1539,8 +1605,15 @@ protected override void InvokeDefaultAction(string path)
                 if (isDirectory)
                 {
                     DirectoryInfo directory = new DirectoryInfo(path);
+                    InodeTracker tracker = null;
+
+                    if (recurse && Context.MyInvocation.BoundParameters.ContainsKey("FollowSymlink"))
```

Comment:
There are two parameter sets for `Get-ChildItem`: `Items` and `LiteralItems`. The `-Recurse` and `-FollowSymlink` parameters are in both parameter sets.

The `Depth` parameter, which is also in both, implies `-Recurse` by setting the `Recurse` property. Should `-FollowSymlink` follow that pattern? That would remove the check of `recurse` here.